test/ReverseProxy.Tests/Service/Proxy/HttpProxyTests.cs

```diff
@@ -622,6 +622,46 @@ public async Task ProxyAsync_RequetsWithBodies_HasHttpContent(string method, str
             Assert.Equal(StatusCodes.Status200OK, httpContext.Response.StatusCode);
         }
 
+        [Theory]
+        [InlineData("HTTP/1.1")]
+        [InlineData("HTTP/2")]
+        public async Task ProxyAsync_RequestWithCookieHeaders(string protocol)
+        {
+            var cookies = new [] { "testA=A_Cookie", "testB=B_Cookie", "testC=C_Cookie" };
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Method = "GET";
+            httpContext.Request.Protocol = protocol;
+            httpContext.Request.Headers.Add("Cookie", cookies.Take(2).ToArray());
+            httpContext.Request.Headers.Append("Cookie", cookies.Last());
```

Comment:
Append aggregates into a single StringValues, this is the same as calling:
```suggestion
            httpContext.Request.Headers.Add("Cookie", cookies);
```

Reply:
Done.